syntax = "proto3";

package helloworld;

service Greeter {

  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SendImage (ImageRequest) returns (ImageReply) {}

  rpc SendVideo (VideoRequest) returns (VideoReply) {}

}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

message ImageRequest {
  bytes image = 1;
}

message ImageReply {
  string message = 1;
}

message VideoRequest {
  bytes video = 1;
}

message VideoReply {
  string message = 1;
}